### PR TITLE
Additional usage examples

### LINKED ISSUE #47: Where is the Pool of connections, and how to limit it.
Hi, I'm trying to migrate https://github.com/suharev7/clickhouse-rs to use yours, but I cannot find where the Pool is.
I want to have some connections always established, and need to keep at least one at all times, i.e. pool_min and pool_max.

Your docs say almost nothing about it, besides that one sentence "Reuse created clients or clone them in order to reuse a connection pool".
Please, care to explain it better? How does a Pool work here?
Thank you.

### PR DESCRIPTION
**Pull request by slvrtrn**
Related: #116 

### Summary

* A new `examples/README` file with a short overview and simple instructions on how to run it.
* Contains the following new examples:
  - session_id
  - query_id
  - custom HTTP headers
  - custom HTTP client (closes #47)
  - ClickHouse Cloud usage with either `rustls-tls` or `native-tls`

### CODE REVIEW

File: examples/clickhouse_cloud.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+use serde::{Deserialize, Serialize};
+use std::env;
+
+// This example requires three environment variables with your instance credentials to be set
+//
+// - CLICKHOUSE_URL (e.g., https://myservice.clickhouse.cloud:8443)
+// - CLICKHOUSE_USER
+// - CLICKHOUSE_PASSWORD
+//
+// Additionally, `rustls-tls` or `native-tls` cargo features should be enabled.
```
Comment (loyd) [inline examples/clickhouse_cloud.rs:13]: Mm, it should be marked via `required_features` in Cargo.toml

File: examples/clickhouse_cloud.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    Ok(())
+}
+
+#[derive(Debug, Serialize, Deserialize, Row)]
+struct Data {
+    id: u32,
+    name: String,
+}
+
+const CLICKHOUSE_URL: &str = "CLICKHOUSE_URL";
```
Comment (loyd) [inline examples/clickhouse_cloud.rs:82]: What's the point of these constants?

Comment (slvrtrn) [reply examples/clickhouse_cloud.rs]: Removed.

File: examples/clickhouse_cloud.rs (hunk 1 above)
Comment (slvrtrn) [reply examples/clickhouse_cloud.rs]: This is great. Added.

File: examples/custom_http_client.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+use hyper_util::client::legacy::connect::HttpConnector;
+use hyper_util::client::legacy::Client as HyperClient;
+use hyper_util::rt::TokioExecutor;
+
+use clickhouse::{error::Result, Client};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let connector = HttpConnector::new(); // or HttpsConnectorBuilder
```
Comment (loyd) [inline examples/custom_http_client.rs:11]: I think we should note users that it's unstable feature and will be broken 100% eventually

**Pull request merged**